lib/spack/spack/cmd/load.py

```diff
@@ -55,7 +55,9 @@ def setup_parser(subparser):
 
 def load(parser, args):
     env = ev.get_env(args, 'load')
-    specs = [spack.cmd.disambiguate_spec(spec, env, first=args.load_first)
+    specs = [spack.cmd.disambiguate_spec(spec, env, first=args.load_first,
+                                         compatible=True,
+                                         preferred=True)
```

Comment:
These options need to be somehow controlled by options. In particular, there are lots of builds that are incompatible *as builds* that are compatible at runtime (different library versions can run together even if they weren't built together. In Spack, we don't try to model that, but we don't want to preclude users from doing it themselves either). I *think* the default behavior should be to load regardless of whether we found a compatible version, and only error out if the user requested that behavior.